UnhollowerBaseLib/ClassInjector.cs

```diff
@@ -18,6 +18,13 @@
 
 namespace UnhollowerRuntimeLib
 {
+
+    public struct RegisterTypeOptions
+    {
+        public bool LogSuccess = true;
+        public Func<Type, Type[]> InterfacesResolver = null;
+    }
```

Comment:
```suggestion
    public class RegisterTypeOptions
    {
        public bool LogSuccess { get; init; } = true;
        public Func<Type, Type[]> InterfacesResolver { get; init; } = null;
    }
```

I suggest using properties here as it will account for possible cases of having to obsolete some fields and have automatic logic for converting them to new ones. 

You could also use a `record` here you'd also get some automatic clean `ToString` implementation for potential logging.

If your idea was to use `struct` to prevent it from escaping to the heap, maybe use `readonly ref struct` instead.